src/buffer_logger/can_buffer_logger.py

```diff
@@ -0,0 +1,81 @@
+import can
+import asyncio
+from buffer_logger.buffer_logger import BufferLogger
+from numpy import array2string, empty, set_printoptions
+from datetime import datetime
+
+
+class CanBufferLogger(BufferLogger):
+    ''' This class is being used to log all 'raw' can messages asychronously.
+    '''
+
+    def __init__(self, config, config_type):
+        super().__init__(config, config_type)
+        can_logging = self.config[self.config_type]['raw_can_data_logging']
+        if(can_logging['enabled']):
+            # Creates new network connection because CANopen library has
+            # difficulties with multithreading on a single network connection.
+            self.can_logging_buffer = can_logging['buffer']
+            self.buffered_data = empty([self.can_logging_buffer],
+                                       dtype=can.Message)
+            set_printoptions(threshold=self.can_logging_buffer)
+
+    def connect_to_network(self):
+        ''' Listens to the configured channel, bustype and bitrate.
+
+            Returns void.
+        '''
+        self.network = can.Bus(
+            self.config[self.config_type]['channel'],
+            bustype=self.config[self.config_type]['bustype'],
+            bitrate=self.config[self.config_type]['bitrate'],
+            receive_own_messages=True)
+
+    async def listen_to_network(self):
+        ''' Firstly, connects to network and creates async listener variables.
+            Then initializes variables needed for to listen to the can
+            messages. And later when while loop is cancelled then stop
+            the listener.
+
+            Returns void.
+        '''
+        self.connect_to_network()
+        self._starting_async_listener()
+        # Uses circular buffer implementation which overwrites
+        # oldest index with new data once the buffer is full.
+        index = 0
+        while True:
```

Comment:
no exit condition/infinite loop

Reply:
In principe wil je dat wel, want het als het CargoDash draait en can berichten komen binnen dan moet je altijd gaan loggen.

ik voeg wel try .. catch KeyboardInterrupt toe.. om het wat netter te maken